chain/consensus/mir/app.go

```diff
@@ -0,0 +1,79 @@
+package mir
+
+import (
+	"sync"
+
+	"github.com/filecoin-project/mir/pkg/modules"
+	"github.com/filecoin-project/mir/pkg/pb/requestpb"
+)
+
+type Tx []byte
+
+type Application struct {
+	mu sync.Mutex
```

Comment:
It depends on the request store and how many clients share the application. If all structures are unique then we don't need this

Reply:
The App abstraction is meant to only be used as a module for a single instance of Mir, so it looks like we don't need to synchronize access to the `Apply` method. This is not to say that the object implementing the `App` interface cannot interact with the rest of the system outside of Mir - it has to, in fact, otherwise it wouldn't be very useful. It's just that the `Apply` function will never be called concurrently. A mutex might still make sense if the `Apply` method accesses some data structures that are also accessed through other methods not called by Mir, but that doesn't seem to be the case now.

Reply:
ok. removed.